config/jest/setupTests.js

```diff
@@ -31,10 +31,12 @@ beforeEach(() => {
 });
 
 afterEach(() => {
-	const actual = console.warn.mock.calls.concat(console.error.mock.calls).filter(([m]) => filterExp.test(m));
+	const actual = (console.warn.mock ? console.warn.mock.calls : [])
+		.concat(console.error.mock ? console.error.mock.calls : [])
+		.filter(([m]) => filterExp.test(m));
 	const expected = 0;
-	console.warn.mockRestore();
-	console.error.mockRestore();
+	if (console.warn.mock) console.warn.mockRestore();
+	if (console.error.mock) console.error.mockRestore();
```

Comment:
Is this test valid when `console.warn.mock` and `console.error.mock` is undefined?
Or should we wrap `if (console.warn.mock && console.error.mock)` condition to whole code block?

Reply:
If developers only use "jest.restoreAllMocks()" in their test scripts, then wraping `if (console.warn.mock && console.error.mock)` is the right choice.
But I choose this way to cover the individual restoring case.
No one knows that the developer will restore whole mocks or some mocks.